org/enigma/EnigmaWriter.java

```diff
@@ -169,14 +171,20 @@ protected void populateStruct() {
 		// TODO: Populate constants from chosen configuration on the main
 		// toolbar.
 
-		// TODO: Fixme
-		o.includeCount = 0;// i.includes.size();
-		/*
-		 * if (o.includeCount != 0) { o.includes = new Include.ByReference();
-		 * Include[] oil = (Include[]) o.includes.toArray(o.includeCount); for
-		 * (int inc = 0; inc < o.includeCount; inc++) { oil[inc].filepath =
-		 * i.includes.get(inc).filepath; } }
-		 */
+		ResourceList<org.lateralgm.resources.Include> includeList = i.resMap.getList(org.lateralgm.resources.Include.class);
+		org.lateralgm.resources.Include iil[] = includeList.toArray(new org.lateralgm.resources.Include[0]);
```

Comment:
Gonna jump on this and note that `toArray(T[] a)` uses the size of the collection, not the array parameter, to determine the size of the returned array. Thus, the 0 is fine and I took that from the sprite/other resource population logic.

> Otherwise, a new array is allocated with the runtime type of the specified array and the **size of this collection**
https://docs.oracle.com/javase/7/docs/api/java/util/AbstractCollection.html#toArray(T[])
https://docs.oracle.com/javase/7/docs/api/java/util/AbstractCollection.html#toArray()